### ISSUE TITLE
documenting a optinionated and powerful developer setup - developer UX

### ISSUE THREAD

**Issue by bgruening** [usegalaxy-au/elixir-biocommons-colab#17]
@davelopez has written up his development setup to kickstart new contributors.

* VSCode Profile Galaxy Development: Contains the minimum extensions that I consider essential (https://gist.github.com/davelopez/ad706e4e35e1d3a20321a2bf717077ac)
* Workspace Settings: This file needs to go inside the .vscode folder of the repository (https://gist.github.com/davelopez/f401f90ed8ed469c69e74442e6a37664)
* Debug Launch Settings: For launching the server (and/or Celery or UI client) in debug mode. Has to be placed in the .vscode folder of the repository too (https://gist.github.com/davelopez/fa49ecb09c0af1c4f092b3c41b1cb1df)

**Comment by neoformit** [comment]
Nice!

**Comment by KaiOnGitHub** [comment]
does this setup automatically recompile after code changes or do you have to restart the server?

**Comment by davelopez** [comment]
If you are referring to the client UI, you can try these commands https://github.com/galaxyproject/galaxy/blob/1eac67ac669a51d9df2748f72053af5fea10cffb/client/README.md?plain=1#L51

For the server, unfortunately, as far as we know, there is no way to auto-reload the changes reliably.

**Comment by KaiOnGitHub** [comment]
@davelopez Thanks for the quick response! That's a pitty, but at least it works for the client. So you make your changes, quit the server and run "sh run.sh" again? This takes quite long for me, is there a better workflow?

**Comment by davelopez** [comment]
Yes, use `make skip-client` to relaunch the server. Running `sh run.sh` will also rebuild the client which is the slowest part. 

So, what I usually do, is run `make client-dev-server` in one terminal and `make skip-client` in another. And when I want to try the changes in the backend I just re-run `make skip-client`.

**Comment by KaiOnGitHub** [comment]
That's great. I just noticed, that it is also rather fast to just use The GalaxyFastAPI uvicorn debug config and this allows me to debug as well then.

Don't remember where I got it from, but this my my config from launch.json for VSC:
```
{
            "name": "GalaxyFastAPI uvicorn",
            "type": "python",
            "request": "launch",
            "module": "uvicorn",
            "args": ["--app-dir", "lib",  "--factory", "galaxy.webapps.galaxy.fast_factory:factory"],
            "env": {
                "GALAXY_CONFIG_FILE": "${workspaceFolder}/config/galaxy.yml"
            }
        },

**Comment by davelopez** [comment]
Sure, if you want to debug that is perfectly fine. You can find a similar launch configuration with some more tweaks in the `Debug Launch Settings` posted by Bjorn https://github.com/usegalaxy-au/elixir-biocommons-colab/issues/17#issue-1821326338 at the top.
